charms/kserve-controller/tests/unit/test_charm.py

```diff
@@ -79,6 +79,12 @@ def mocked_gen_certs(mocker):
     yield mocker.patch("charm.KServeControllerCharm.gen_certs")
 
 
+@pytest.fixture()
```

Comment:
Well, it's either this or add a decorator on all test cases that require this mocked method. Another alternative is to mock away the method inside the test case, which is inconvenient because it's the same code repeated in multiple parts.

Reply:
I think it should be a decorator since we do not use fixture

or alternative module scoped fixture with `autouse=True`, it that actually possible

Reply:
great, thanks!